### ISSUE TITLE
[Bug]: undoInputRule almost never works

### ISSUE THREAD

**Issue by Slapbox** [ueberdosis/tiptap#4413]
### Which packages did you experience the bug in?

core

### What Tiptap version are you using?

2.1.8

### What’s the bug you are facing?

`undoInputRule` almost never has any effect and I can't find any pattern to when it does.

Determining the correct behavior of this command would be much easier with an example. I'd assume that it would undo the markdown conversion of `-` into a list, but instead it returns false, and it does the same for every input rule - but I have accidentally seen it work once or twice while working on other tasks.

### What browser are you using?

Chrome

### What did you expect to happen?

Undoes most recent input rule as described, but instead nothing happens and it returns `false`

### Anything to add? (optional)

It seems to expect a plugin's `state` to tell it whether there's an input rule to undo but what plugin in TipTap actually tracks that in `state`?

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by Slapbox** [comment]
My fault!

**Comment by mejo-** [comment]
@Nantris I experience exactly what you described in this issue. Would you mind to share what your fault was, as I can imagine that I have the same 😆

**Comment by Nantris** [comment]
Oh goodness I'm sorry, it's been so long I haven't the faintest clue.

**Comment by mejo-** [comment]
No worries @Nantris, we figured it out.

In our case it was a ProseMirror plugin that altered nodes in the document via `appendTransaction` after each transaction that implied a document change. We adjusted the plugin now in two ways:
1. Only touch the actually changed nodes of the transaction and only apply changes to the nodes if necessary.
2. Ignore transactions that have the `InputRulePlugin` state set.
